src/components/ServiceList.jsx

```diff
@@ -1,22 +1,58 @@
 import { Alert } from "@baltimorecounty/dotgov-components";
-import FilterList from "../components/FilterList";
-import { MostPopularServiceIconStyles } from "../styles";
-import React from "react";
-import ServiceIconLink from "../components/ServiceIconLink";
+import FilterList from "./FilterList";
+import React, { useState } from "react";
+import ServiceIconLink from "./ServiceIconLink";
+import PopularityFilterCollapse from "./PopularityFilterCollapse";
 import useServices from "../hooks/useServices";
+import ListLegend from "./ListLegend";
+import ListCounter from "./ListCounter";
+import { TextInput } from "@baltimorecounty/dotgov-components";
+
+const filterByTextInput = (services, searchText) => {
+  return services.filter(item => {
+    const { name, department } = item;
+    return (
+      name.toLowerCase().indexOf(searchText.toLowerCase()) !== -1 ||
+      department.toLowerCase().indexOf(searchText.toLowerCase()) !== -1
+    );
+  });
+};
+
+const filterByPopularity = item => item.rank > 0;
 
 const ServiceList = () => {
   const { hasError, serviceItems = [], isLoading } = useServices();
+  const [isMostPopular, setMostPopular] = useState(false);
+  const [filteredItems, setFilteredItems] = useState([]);
+  const [filterText, setFilterText] = useState([]);
+  const [isFiltering, setIsFiltering] = useState(false);
+
+  const filterServiceList = (checkedVal, searchText) => {
+    let checkedItems = [];
```

Comment:
Ok this is starting to make more sense, but i suspect this can be cleaned up.

There are a lot of conditionals in here, we can simplify using [array.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) and chaining.

Proposed 

```js
const items = [...serviceItems]
      .filter(item => !shouldShowMostPopularServices || item.rank > 0)
      .filter(item => !searchText.trim() || filterByText(item, searchText));

setFilteredItems(items);
```

If we move this line by line:

### Service Items

```js
const items = [...serviceItems]
```

Creates a copy of the existing services list, so we don't impact the existing array 

### Most Popular

```js
  .filter(item => !shouldShowMostPopularServices || item.rank > 0)
```

This line addresses whether a services is popular (checkedVal in the original code), if this value is false, then we will return all service items, otherwise we use our most popular condition.

### Filter Text

```js
.filter(item => !searchText.trim() || filterByText(item, searchText));
```
Similar to the most popular filter...If there isnt' a search term give us everything, otherwise filter by the search term.  `filterByText` is similar to the existing `filterByTextInput` but only addresses a single item.

This approach will allow us to easily add other filters into this logic by just chaining another filter statement.

Let me know if this makes sense, sorry for the novel.

Reply:
This is done